OSDOCS#6494: vSphere - move install config parameters out of assemblies

mjpytlak opened the pull request:
Version(s):
4.13+

Issue:
This PR addresses [osdocs-6494](https://issues.redhat.com/browse/OSDOCS-6494)/[osdocs-6493](https://issues.redhat.com/browse/OSDOCS-6493)

Link to docs preview:

- [Installation configuration parameters](https://61165--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installation-config-parameters-vsphere.html) 
New assembly at book level
- Installing a cluster on vSphere with customizations > [Creating the installation configuration file](https://61165--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html#installation-initializing_installing-vsphere-installer-provisioned-customizations)
Note the absence of Installation configuration parameters following this topic. In liue of the module, a cross reference is provided for the new assembly.

It is worth noting that the latter is an example of the change that this PR introduces. This change has been applied to all vSphere installation assemblies. 

QE review:
- [NA ] QE has approved this change.
See conversation below.

Also worth noting that this work can only be merged into 4.14. A back port to 4.13 requires a separate PR to prevent FIPS content from appearing in the docs.

skrthomas reviewed (commented):
Nice work and definitely a helpful modular change :) LGTM

mjpytlak commented:
@kalexand-rh I believe the answer is yes, but just looking to confirm that QE is not required for this improvement. I moved the installation configuration parameters out of each vSphere assembly and added them at the book level. No content was changed.

sjstout commented:
LGTM! Thanks for taking this on, Mike!

mjpytlak commented on installing/installing_vsphere/installation-config-parameters-vsphere.adoc:
```diff
@@ -0,0 +1,11 @@
+
+toc::[]
+
+Before you deploy an {product-title} cluster, you provide parameter values to describe your account on the cloud platform that hosts your cluster and optionally customize your cluster's platform. When you create the `install-config.yaml` installation configuration file, you provide values for the required parameters through the command line. If you customize your cluster, you can modify the `inst ...
```
@kalexand-rh That is because it is not just vSphere installation parameters. :smile: 

The module is comprised of required and optional parameters that are platform agnostic, plus platform specific parameters. In this case, vSphere.  Thanks for pointing this out, however, as I think this refactor should point out that the tables represent required, optional, a platform specific parameters.

mjpytlak replied:
@kalexand-rh Oh I see what you mean now. I will update the abstract to specify that these are installation parameters used to install a cluster on vSphere.

mjpytlak commented on _topic_maps/_topic_map.yml:
```diff
@@ -504,6 +504,8 @@ Topics:
     File: uninstalling-cluster-vsphere-installer-provisioned
   - Name: Using the vSphere Problem Detector Operator
     File: using-vsphere-problem-detector-operator
+  - Name: Installation configuration parameters
```
Yep agreed. In fact, Stephanie had the same request when I presented this solution during an installation sync call.

mjpytlak commented:
/cherrypick enterprise-4.14

SNiemann15 commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -231,26 +220,58 @@ endif::[]
+// This accommodates the existing IA of the installation assemblies, while the improvement is implemented.
+// As part of the updates for the last provider, the conditions can be removed and the following heading can be used.
+ifdef::vsphere[]
+= Available installation configuration parameters for {platform}
```
@mjpytlak here
